site/docs/ch/optimizing.md

```diff
@@ -1,39 +1,97 @@
-# Optimizers
+# Optimizing
 
-## 概念
+结构优化（Optimize）功能为 Arctic 表提供了自动化的文件治理能力，主要面向解决小文件积压和数据冗余问题。
 
-### Optimizer
+小文件积压主要出现在实时数据写入的场景：为了保证数据写入的实效性，需要进行频繁的数据提交，从而产生大量的小文件，积压的小文件一方面会影响数据的查询性能，另一方面也会对文件系统带来压力。
 
-Optimizer 是结构优化（Optimize）功能的执行组件，结构优化时数据的转化、合并等文件读写操作，都依赖 Optimizer 来完成。
+数据冗余主要出现在数据更新的场景：为了避免数据更新时频繁重写文件，造成写放大问题，引入了 delete 文件对数据进行标记删除，而不断积累的 delete 文件会造成数据冗余，不仅占用磁盘空间，也会影响查询性能。
 
-Optimizer 的功能比较单一，只负责结构优化功能 plan、execute、commit 三个步骤中的 execute 部分。
+Arctic 引入了 Optimize 功能来解决上述问题，通过文件的移动、转化、合并等操作满足文件合并、数据去重、数据排序等需求，Arctic 的 Optimize 主要有如下几个核心特点：
 
-具体来说，Optimizer 需要完成从 AMS 获取待执行的任务、执行任务、将任务执行结果上报给 AMS 三个步骤。
+- 自动化流式执行：后台任务持续监听文件变化，异步执行结构优化
 
-每个 Optimizer 都是一个实际运行的服务，它的运行方式不受限制，可以是一个本地进程，也可以是运行在分布式资源管理系统（如 yarn）上的作业。
+- 资源隔离和共享：允许资源在表级隔离和共享，以及设置资源配额
 
-### Optimizer Container
+- 灵活的部署方式：执行节点支持多种部署方式，便捷的扩缩容
 
-Optimizer Container（Optimizer 容器）是 Optimizer 的运行容器，用来解决 Optimizer 的在不同运行环境下的加载、停止等调度问题。
+## Optimize 的分类
 
-Optimizer Container 可以有多种类型（type），Arctic 已经提供了 local 和 flink 两种类型的实现，分别支持 Optimizer 运行在本地环境和 Flink 集群。
+从功能上划分为三类：[Minor Optimize](#minor-optimize)、[Major Optimize](#major-optimize) 和 [Full Optimize](#full-optimize)。
 
-Optimizer Container 本身不是服务，它只代表了一组具体的运行环境配置，以及在该运行环境下 Optimizer 的调度方案。
+其中 Minor Optimize 只合并小文件，超过小文件大小阈值的文件一般不再参与 Minor Optimize，其主要目标是削减小文件数量，进而带来查询性能的提升；
 
-### Optimizer Group
+Major Optimize 将未达到到目标大小的数据文件以及 delete 文件进行合并，生成接近理想目标大小的文件，主要目标是提升平均文件大小到接近目标值、以及减少冗余数据，进而进一步提升查询性能以及释放磁盘空间；
 
-Optimizer Group（Optimizer 资源组）是为了划分 Optimizer 资源而引入的概念，一个 Optimizer Group 包含了若干个属于同一资源池的 Optimizer 服务。
+Full Optimize 一次性触发所有文件的合并，达到彻底的文件规整、消除冗余数据的目标。
 
-启动一个 Optimizer 服务，必须指定其所属的 Optimizer Group，这也限定了一个 Optimizer 服务只能属于一个 Optimizer Group。
+### Minor Optimize
 
-创建 Optimizer Group 需要指定它的 Optimizer Container，即定义这个 Optimizer Group 下的 Optimizer 的运行环境，当前一个 Optimizer Group 只能设置一个运行环境。
+Minor Optimize 只处理小文件，Arctic 支持为每张表独立配置小文件的阈值，文件大小低于阈值的文件即小文件。
+
+![Minor Optimize](images/minor-optimize.png)
+
+在合并小文件时，与这些小文件相关的 delete 文件也会参与，而被 delete 文件删除的数据不会输出到合并后的文件中；并且，如果 delete 文件只和这些小文件相关，合并完成后，delete 文件也会被清理。
+
+由于只处理小文件，Minor Optimize 只能将 Arctic 表的平均文件大小提升到一个比较低的值，但是带来的好处是，Minor Optimize 的执行代价相对较低，因此执行频率可以更加激进一些，一般可以配置为几分钟到几十分钟。
+
+需要特殊说明的是，在 Arctic 有主键表场景下，Minor Optimize 不仅负责小文件的合并，也负责 Changestore 到 Basestore 的数据迁移。
+
+![KeyedTable Minor Optimize](images/keyed-table-minor-optimize.png)
+
+Changestore 中的文件都被视作小文件参与 Minor Optimize，包括：
+
+- 将 Changestore 中的 insert 文件合并到 Basestore 中
+
+- 将 Changestore 中的 eq-delete 文件合并为 Basestore 中的 pos-delete 文件（需要读取 Basestore 中的数据）
+
+对有主键表来说，Minor Optimize 不仅削减了小文件数量，也提升了 Basestore 的数据实效性，同时由于 eq-delete 转化成了 pos-delete，对查询性能的提升也有帮助。
+
+### Major Optimize
+
+Major Optimize 的核心目标是提升平均文件大小、清理冗余数据。
```

Comment:
what is the difference between minor and major on file sizing? this is somehow ambiguous